include/libcyphal/transport/msg_sessions.hpp

```diff
@@ -0,0 +1,59 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_MSG_SESSIONS_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_MSG_SESSIONS_HPP_INCLUDED
+
+#include "session.hpp"
+
+namespace libcyphal
+{
+namespace transport
+{
+
+struct MessageRxParams final
+{
+    std::size_t extent_bytes;
+    PortId      subject_id;
+};
+
+struct MessageTxParams final
+{
+    PortId subject_id;
+};
+
+class IMessageRxSession : public IRxSession
+{
+public:
+    CETL_NODISCARD virtual MessageRxParams getParams() const noexcept = 0;
+
+    /// @brief Receives a message from the transport layer.
+    ///
+    /// Method is not blocking, and will return immediately if no message is available.
+    ///
+    /// @return A message transfer if available; otherwise an empty optional.
+    ///
+    CETL_NODISCARD virtual cetl::optional<MessageRxTransfer> receive() = 0;
+};
+
+class IMessageTxSession : public IRunnable
```

Comment:
This is sensible but:

- Sometimes it is necessary to provide more holistic documentation covering multiple entities at once instead of scattering it per entity. This is one such case. We could copy-paste relevant parts of the design doc to the `ITransport` docs in `transport.hpp` and put references to that in the class comments here. The relevant sections are:
   - https://github.com/OpenCyphal-Garage/libcyphal/blob/main/docs/design/readme.md#session-factory-methods
   - https://github.com/OpenCyphal-Garage/libcyphal/blob/main/docs/design/readme.md#session-data-flow

- Some methods do not require documentation. If you have `Thing getThing() const noexcept`, a doc comment saying "Gets the thing" adds nothing but noise. I would prefer to keep obvious cases undocumented.

Reply:
As, I wrote for other PR, documentation for public stuff is definitely coming, but I believe we first need some more or less stable api agreed upon. Design doc is very good and very insightful, but at the same time sometimes leaves some (devil in) details unspecified, which might affect result interfaces/implementation quite significantly (like we already discussing about error handling strategies).

Reply:
It's mostly the class-level documentation that I'm struggling with. I'm left to linking your intentions in these PRs with a single word – the class name – against the design document. If you have to c&p from that document it's fine just help me out with some context when you first define a class, interface, or data structure.